### PR TITLE
Fixed Helmholtz example - Neumann Boundary Condition

### PR DESCRIPTION
**Pull request by blaszm**
The weak form formulation contained an error with the Neumann Boundary condition, which should vanish because of partial integration. Addtionally, the sign of g when added to the local force vector was incorrect. We fixed both these errors and also the documentation and the test at the end. 

Please note that the influence of this change is not directly visible in ParaView, as the used analytical function has three peaks in the middle of the region, thus the boundary condition has barely any influence. By changing the center to xs = (Vec{2}((-0.95,  0.95)),          Vec{2}((-0.95, -0.95)), Vec{2}(( 0.5,  -0.5))), the effect can be seen:
PV_helmholtz

Here, the top example is the old implementation, which shows oscillations in the bottom and a deformation in the top left corner, while the improved version looks much smoother and as one would expect the solution to look like.

### CODE REVIEW

**Review by termi-official — APPROVED** [review]
LGTM.

File: docs/src/literate/helmholtz.jl [hunk 1]
```diff
@@ -26,7 +26,7 @@
 # \int \nabla δu \cdot \nabla u d\Omega
 # + \int δu \cdot u d\Omega
 # - \int δu \cdot f d\Omega
-# + \int δu \cdot (n \cdot \nabla u - g_2) d\Gamma_2
+# - \int δu \cdot g_2 d\Gamma_2 = 0 \forall δu
```
Comment (kimauth) [inline docs/src/literate/helmholtz.jl:29]: I think there is no need for the \cdot here, is there? Seems to me like these are all scalars (same for the two terms above).

Comment (kimauth) [reply docs/src/literate/helmholtz.jl:29]: Maybe it would be easier to read if it was `int_{\Gamma_2}` as well (and similar for the other terms), once we're on changing it anyways. What do you think?

**Review by kimauth — COMMENTED** [review]
Good catch, thank you!
